jax/_src/core.py

```diff
@@ -3152,8 +3152,15 @@ def suggest_same_var_names(self,
         self.var_names[for_v] = pp_var(like_v, self)
 
 
-def pp_var(v: Var | Literal, context: JaxprPpContext) -> str:
-  if isinstance(v, (Literal, DropVar)): return str(v)
+def pp_var(v: Var | Literal, context: JaxprPpContext, *,
```

Comment:
Why not change the str_short of `Literal` instead?

Reply:
Because of the change in `state/indexing.py`.

Reply:
Why do we need that change in indexing.py though?

Reply:
Otherwise we end up with things like

```
b:i32[1] <- a[:,0:i32]
```

instead of

```
b:i32[1] <- a[:,0]
```

when printing refs.

Your original version of the change produced:

```
b:i32[1] <- a[:,0:i32[]]
```

which was definitely horrible! That's why I made this change.

I'll admit that the version without the empty `[]` perhaps isn't so bad.

Reply:
I guess it's actually still bad because it looks like a slice!

Reply:
Ahh, I see. Okay sounds good.

But I still feel that this should be controlled by `str_short` on Literal. What I mean is we default to printing `dtype` in `str_short` of Literal and then in pp_var, you can do this:

```
if isinstance(v, Literal):
  return str(v) if print_literal_dtype else v.str_short(False)
```

Then you can remove the `pp_var` logic in core.py.

WDYT?

Reply:
SGTM! I'll do that.

Reply:
Done. Let me know what you think!